src/spaces/utilities.jl

```diff
@@ -226,12 +228,169 @@ end
 Invoke a random walk by providing the `rand` function in place of
 `direction`. For `AbstractGridSpace`, the walk will cover ±1 positions in all directions,
 `ContinuousSpace` will reside within [-1, 1].
+
+This functionality is deprecated. Use `randomwalk!` instead.
 """
-walk!(agent, ::typeof(rand), model::ABM{<:AbstractGridSpace{D}}; kwargs...) where {D} =
+function walk!(agent, ::typeof(rand), model::ABM{<:AbstractGridSpace{D}}; kwargs...) where {D}
+    @warn "Producing random walks through `walk!` is deprecated. Use `randomwalk!` instead."
     walk!(agent, Tuple(rand(model.rng, -1:1, D)), model; kwargs...)
+end
 
-walk!(agent, ::typeof(rand), model::ABM{<:ContinuousSpace{D}}) where {D} =
+function walk!(agent, ::typeof(rand), model::ABM{<:ContinuousSpace{D}}) where {D}
+    @warn "Producing random walks through `walk!` is deprecated. Use `randomwalk!` instead."
     walk!(agent, Tuple(2.0 * rand(model.rng) - 1.0 for _ in 1:D), model)
+end
+
+"""
+    randomwalk!(agent, model::ABM{<:AbstractGridSpace}, r; kwargs...)
+Move `agent` for a distance `r` in a random direction respecting boundary conditions
+and space metric.
+For Chebyshev and Manhattan metric, the step size `r` is rounded to `floor(Int,r)`;
+for Euclidean metric in a GridSpace, random walks are not defined due to a strong
+dependency on the given value of `r`.
+
+## Keywords
+- `ifempty` will check that the target position is unoccupied and only move if that's true.
+  By default this is true, set it to false if different agents can occupy the same position.
+  In a GridSpaceSingle, agents cannot overlap and this keyword has no effect.
+"""
+function randomwalk!(
+    agent::AbstractAgent,
+    model::ABM{<:AbstractGridSpace},
+    r::Real;
+    kwargs...
+)
+    if model.space.metric == :euclidean
+        throw(ArgumentError(
+            "Random walks on a `GridSpace` with Euclidean metric are not defined. " *
+            "You might want to use a `ContinuousSpace` or a different metric."
+        ))
+    end
+    offsets = offsets_at_radius(model, r)
+    walk!(agent, rand(offsets), model; kwargs...)
```

Comment:
Ops @mastrof I just now saw this!!! Very important to fix: In all calls to `rand` you must use the model rng!!! Please replace all `rand(x)` with `rand(model.rng, x)` everywhere in the code you've added!!!

Reply:
should I force push or pull from main?

Reply:
first pull, as I added a merge commit for the version in project.toml. then edit, and then pull. or merge in your local, then push.